### PR TITLE
CAMEL-14780 camel-undertow: Enhance SPI API for HttpHandler change

### PR DESCRIPTION
**Pull request by JiriOndrusek**
Issue: https://issues.apache.org/jira/browse/CAMEL-14780

Change adds method co change httpHandler into SPI API.
Extension of API is back compatible with previous version of API.
Added JUnit test validating htppHandler 'hook' and test for usage of security parameters defined on component.

[ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
[ ] Each commit in the pull request should have a meaningful subject line and body.
[ ] If you're unsure, you can format the pull request title like `[CAMEL-XXX] Fixes bug in camel-file component`, where you replace `CAMEL-XXX` with the appropriate JIRA issue.
[ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
[ ] Run `mvn clean install -Psourcecheck` in your module with source check enabled to make sure basic checks pass and there are no checkstyle violations. A more thorough check will be performed on your pull request automatically.
Below are the contribution guidelines:
https://github.com/apache/camel/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowComponent.java [hunk 1]
```diff
@@ -78,6 +87,7 @@
     @Metadata(label = "security")
     private String allowedRoles;
 
+    private UndertowSecurityProvider securityProvider;
```
Comment (davsclaus) [inline components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowComponent.java]: Add @Metadata with security label.

File: components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowComponent.java [hunk 2]
```diff
@@ -435,4 +476,8 @@ public String getAllowedRoles() {
     public void setAllowedRoles(String allowedRoles) {
         this.allowedRoles = allowedRoles;
     }
+
+    public UndertowSecurityProvider getSecurityProvider() {
```
Comment (davsclaus) [inline components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowComponent.java]: Add setter so you can set this explicit

File: components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowComponent.java [hunk 3]
```diff
@@ -306,6 +339,8 @@ public Producer createProducer(CamelContext camelContext, String host,
     protected void doStart() throws Exception {
         super.doStart();
 
+        initSecurityProvider();
```
Comment (davsclaus) [inline components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowComponent.java]: Only init if the security provider is null

File: components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowEndpoint.java [hunk 4]
```diff
@@ -147,7 +147,11 @@ public UndertowComponent getComponent() {
     }
 
     public UndertowSecurityProvider getSecurityProvider() {
-        return securityProvider;
+        if (this.securityProvider == null) {
```
Comment (davsclaus) [inline components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowEndpoint.java]: getter/setters should be plain, not these tricks. Instead when creating the endpoint in the component, then set the security provider on the endpoint correctly

File: components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowEndpoint.java [hunk 5]
```diff
@@ -454,14 +458,13 @@ public boolean isPreserveHostHeader() {
...
+        return this.securityConfiguration;
     }
 
     public void setSecurityConfiguration(Object securityConfiguration) {
         this.securityConfiguration = securityConfiguration;
     }
 
-
     public String getAllowedRoles() {
         return allowedRoles == null ? getComponent().getAllowedRoles() : allowedRoles;
```
Comment (davsclaus) [inline components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowEndpoint.java]: See prev comment, getter/setter should be plain

File: components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowComponent.java (hunk 1 above)
Comment (JiriOndrusek) [reply components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowComponent.java]: @davsclaus this is not parameter which comes from configuration, it is just private variable that is used by component when SPI implementation is searched for. From my POV there should be annotation like @nometadata in tis case.

File: components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowComponent.java (hunk 2 above)
Comment (JiriOndrusek) [reply components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowComponent.java]: I can do it also this way, in that case please ignore my previous comment

File: components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowEndpoint.java (hunk 4 above)
Comment (JiriOndrusek) [reply components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowEndpoint.java]: ok, I'll do it, thanks for explanation

File: components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowEndpoint.java (hunk 5 above)
Comment (JiriOndrusek) [reply components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowEndpoint.java]: I'll change it

File: components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowComponent.java (hunk 3 above)
Comment (JiriOndrusek) [reply components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowComponent.java]: I'll change it

**Comment by JiriOndrusek** [comment]
@oscerd I've seen the template, I'll read it, validated all steps. Should it be used in any other way?

**Comment by oscerd** [comment]
There was a contribution about that, but I'm not really sure is useful, too much text.. People may lose attention

**Comment by omarsmak** [comment]
@oscerd I'd say we keep it for sometime and see if people will respond to it :p. But yeah maybe we can reduce it a bit, an example a [beam PR](https://github.com/apache/beam/pull/11216)

**Comment by JiriOndrusek** [comment]
@davsclaus All requested changes added

**Pull request merged**